server/filestore.go

```diff
@@ -2198,7 +2203,11 @@ func (fs *fileStore) expireMsgsOnRecover() {
 			// Update fss
 			// Make sure we have fss loaded.
 			mb.removeSeqPerSubject(sm.subj, seq)
-			fs.removePerSubject(sm.subj)
+			if fs.removePerSubject(sm.subj) {
+				mb.mu.Unlock()
+				fs.systemTombstoneIfNeeded(sm, "MaxAge")
```

Comment:
Why do we ignore the func return param here?

Reply:
At least we should place a comment if we do mean to ignore returned func and not call it.